### ISSUE TITLE
No predictions from setfit

### ISSUE THREAD

**Issue by simran-padam** [huggingface/setfit#340]
I ran SetFitTrainer to train the model with 5 classes (used multi_target_strategy="one-vs-rest"). My target variable was one hot encoded. When I get my predictions on the test data, I observe for most of the rows (close to 40% of test data), I have no class predictions meaning it is not predicting any class.

Has any experienced this issue before and can provide any suggestions.

Thanks

**Comment by kgourgou** [comment]
Hello!

That's interesting. Do you have any small example that reproduces the issue or do you think it's specific to your data?

**Comment by kgourgou** [comment]
Yeah, very weird. May I ask which version of SetFit you are using? Is it the latest? 

Also, what does the loss look like during training? Is that also weird?

**Comment by kgourgou** [comment]
> Doesn't Setfit uses softmax on the top layer, it is strange it did not predict any class.

Missed this question. Vanilla SetFit uses a scikit-learn LogisticRegression model on top of the sentence-transformer embeddings.
